fmriprep/interfaces/images.py

```diff
@@ -355,70 +392,70 @@ def _run_interface(self, runtime):
         out_report = os.path.abspath('report.html')
 
         # Retrieve xform codes
-        sform_code = img.header._structarr['sform_code']
-        qform_code = img.header._structarr['qform_code']
+        sform_code = int(img.header._structarr['sform_code'])
+        qform_code = int(img.header._structarr['qform_code'])
 
         # Check qform is valid
-        valid_qform = True
-        old_qform_code = int(qform_code)
+        valid_qform = False
         try:
             img.get_qform()
+            valid_qform = True
         except ValueError:
-            valid_qform = False
-            qform_code = nb.nifti1.xform_codes['unknown']
-
-        # Check affine information is valid
-        valid_codes = (qform_code, sform_code) != (0, 0)
+            pass
 
         # Matching affines
         matching_affines = valid_qform and np.allclose(img.get_qform(), img.get_sform())
 
-        # Both okay -> do nothing, empty report
-        if valid_qform and valid_codes and matching_affines:
+        # Both match, qform valid (implicit with match), codes okay -> do nothing, empty report
+        if matching_affines and qform_code > 0 and sform_code > 0:
             self._results['out_file'] = self.inputs.in_file
             open(out_report, 'w').close()
             self._results['out_report'] = out_report
             return runtime
 
+        # A new file will be written
         out_fname = fname_presuffix(self.inputs.in_file, suffix='_valid', newpath=runtime.cwd)
         self._results['out_file'] = out_fname
 
-        warning = False
-        warning_txt = ''
-        description = ''
-        # Fix sform if needed
-        if not valid_codes:
-            # Nibabel derives a default LAS affine from the shape and zooms
-            # Use scanner xform code to indicate no alignment has been done
-            img.set_sform(img.affine, nb.nifti1.xform_codes['scanner'])
-            warning = True
-            warning_txt = 'Invalid header'
-            description += """\
-<p class="elem-desc">Input file does not have valid qform or sform matrix.
-  A default, LAS-oriented affine has been constructed.
-  A left-right flip may have occurred.
-  Analyses of this dataset MAY BE INVALID.
+        # Row 2:
+        if valid_qform and qform_code > 0 and sform_code == 0:
+            img.set_sform(img.get_qform(), qform_code)
+            warning_txt = 'Note on orientation: sform matrix overwritten'
+            description = """\
+<p class="elem-desc">The sform has been copied from qform.</p>
+"""
+        # Rows 3-4:
+        # Note: if qform is not valid, matching_affines is False
+        elif sform_code > 0 and (not matching_affines or qform_code == 0):
+            img.set_qform(img.get_sform(), sform_code)
+            warning_txt = 'Note on orientation: qform matrix overwritten'
```

Comment:
I think overwritten is reasonable here, though, as we may be overwriting a valid qform. (And overwriting a qform feels less worrisome to me than overwriting an sform.)